
Use the code common to driver and utilities from a library defined as a
shared object instead of static linking it into each of them.

--- tries/dump/Makefile.am.orig	2018-09-15 07:03:24.000000000 +0000
+++ tries/dump/Makefile.am	2021-12-13 09:50:07.767518207 +0000
@@ -24,4 +24,4 @@
 dist_man8_MANS = dumpexfat.8
 dumpexfat_SOURCES = main.c
 dumpexfat_CPPFLAGS = -I$(top_srcdir)/libexfat
-dumpexfat_LDADD = ../libexfat/libexfat.a
+dumpexfat_LDADD = ../libexfat/libexfat.la
--- tries/fsck/Makefile.am.orig	2018-09-15 07:03:24.000000000 +0000
+++ tries/fsck/Makefile.am	2021-12-13 09:57:07.813581856 +0000
@@ -24,7 +24,7 @@
 dist_man8_MANS = exfatfsck.8
 exfatfsck_SOURCES = main.c
 exfatfsck_CPPFLAGS = -I$(top_srcdir)/libexfat
-exfatfsck_LDADD = ../libexfat/libexfat.a
+exfatfsck_LDADD = ../libexfat/libexfat.la
 
 install-exec-hook:
 	ln -sf $(sbin_PROGRAMS) $(DESTDIR)$(sbindir)/fsck.exfat
--- tries/fuse/Makefile.am.orig	2018-09-15 07:03:24.000000000 +0000
+++ tries/fuse/Makefile.am	2021-12-13 10:00:21.191369481 +0000
@@ -25,7 +25,7 @@
 mount_exfat_fuse_SOURCES = main.c
 mount_exfat_fuse_CPPFLAGS = -I$(top_srcdir)/libexfat
 mount_exfat_fuse_CFLAGS = $(FUSE_CFLAGS)
-mount_exfat_fuse_LDADD = ../libexfat/libexfat.a $(FUSE_LIBS)
+mount_exfat_fuse_LDADD = ../libexfat/libexfat.la $(FUSE_LIBS)
 
 install-exec-hook:
 	ln -sf $(sbin_PROGRAMS) $(DESTDIR)$(sbindir)/mount.exfat
--- tries/label/Makefile.am.orig	2018-09-15 07:03:24.000000000 +0000
+++ tries/label/Makefile.am	2021-12-13 10:00:50.340717795 +0000
@@ -24,4 +24,4 @@
 dist_man8_MANS = exfatlabel.8
 exfatlabel_SOURCES = main.c
 exfatlabel_CPPFLAGS = -I$(top_srcdir)/libexfat
-exfatlabel_LDADD = ../libexfat/libexfat.a
+exfatlabel_LDADD = ../libexfat/libexfat.la
--- tries/libexfat/Makefile.am.orig	2018-09-15 07:03:24.000000000 +0000
+++ tries/libexfat/Makefile.am	2021-12-13 10:35:05.888333041 +0000
@@ -20,8 +20,8 @@
 #	51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 #
 
-noinst_LIBRARIES = libexfat.a
-libexfat_a_SOURCES = \
+lib_LTLIBRARIES = libexfat.la
+libexfat_la_SOURCES = \
 	byteorder.h \
 	cluster.c \
 	compiler.h \
--- tries/mkfs/Makefile.am.orig	2018-09-15 07:03:24.000000000 +0000
+++ tries/mkfs/Makefile.am	2021-12-13 10:01:12.110173499 +0000
@@ -39,7 +39,7 @@
 	vbr.c \
 	vbr.h
 mkexfatfs_CPPFLAGS = -I$(top_srcdir)/libexfat
-mkexfatfs_LDADD = ../libexfat/libexfat.a
+mkexfatfs_LDADD = ../libexfat/libexfat.la
 
 install-exec-hook:
 	ln -sf $(sbin_PROGRAMS) $(DESTDIR)$(sbindir)/mkfs.exfat
